### DISCUSSION TITLE
Looking for suggestions and C&C

### DISCUSSION THREAD

**Discussion by rbdavid** [BradyAJohnston/MolecularNodes#449] [General]
Hi all, 
I've been doing some initial renderings for a large animation that I've got rattling in my brain. The first is a rotation around an Alphafold2 model (https://youtu.be/cGb0Ug51ph8). The second is a visualization of alignment between the model and an experimental structure (https://youtu.be/025U8be6ejc; this one's a very quick version of the final animation). I haven't added any audio or text to these since they are just initial renders. 

One issue I'm having is that rendering a single frame took around 33 seconds, on average. And these two videos are hundreds of frames. I am using fairly old hardware (NVIDIA GeForce GTX 1080 Ti) with Cycles - GPU Compute for the rendering engine. I haven't really messed with default rendering parameters. Does the community have any suggestions for parameters/strategies to efficiently render high-quality images? 

Any critiques and suggestions are welcome. Thanks!

**Comment by BradyAJohnston** [comment]
The links I think are private as I get 'Video Unavailable'. 

RE: rendering speed. It depends a lot for the intended audience or use case for your render, but I'll list some strategies below that I follow.

1. Reducing resolution and samples. This is an obvious one, but it's something that you can get away with more than you think. For most stuff I post to the internet or use in presentations that I do - I render at 720p (1280x720). As long as the bitrate of the video file is good, 720p actually doesn't look too bad, especially when you are giving talks on a projector that is probably not great to begin with (and it can be more favorable for online compression algorithms as well). Sample counts when doing test renders I usually set it at ~16 samples (really noisy but you can preview stuff), and for 'final renders' I usually don't go above 128 or 256 samples. The noise amount is pretty good, and when combined with the denoising it works really well.
2. Materials used in the scene. If you use complicated materials with effects such as subsurface scattering, this _drastically_ increases render times. If I am wanting fast render times I ensure that my materials are basic. For the fastest render times I use materials that are just ambient occlusion / color, which doesn't even use lighting from the scene. These are super fast and don't look too bad most of the time - especially when rendering spheres for atoms.
3. The more lights you have in a scene the slower it is to render. The same with using HDRIs. The default Molecular Nodes startup file using an HDRI. Disabling this and using a simpler lighting setup can also speed up render times. You can also manually tweak the number of light bounces in cycles. Reducing them all will speed up render times, but lighting will look worse. It will vary from scene to scene, depending on lighting and materials, but tinkering with them and reducing counts to workable amounts will also speed up rendering. Lots of tutorials on youtube on how to do this kind of speedups.
4. Using a render farm. There a bunch of different paid render farm services. I haven't used any personally, but I have used SheepIt render farm. You donate your computers compute time when you aren't using it and gain credits, which you can then spend to submit your own render jobs. It's a 'Free' render farm that is community driven and you pay for by having your own machine do rendering for them as well. A couple of days leaving your computer overnight doing rendering will give you loads of credits to start submitting some jobs to it.
5. If you have access to any other compute resources such as a HPC etc, then you can setup your scene on your personal computer then upload the `.blend` and render elsewhere, via the command line or similar.

**Reply by rbdavid** [reply]
Videos should be public now. Thanks for the suggestions! The default sample value is something like 4096, so I certainly have been overkilling on that count.

**Comment by alphataubio** [comment]
> 5. If you have access to any other compute resources such as a HPC etc, then you can setup your scene on your personal computer then upload the `.blend` and render elsewhere, via the command line or similar.

awesome idea, thanks !

i have access to cluster nodes with 48 cores and 4 x a100's, does rendering work in parallel across CPUs and/or GPUs  ?
